src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -291,6 +291,18 @@ stepWithAxiom
         returnExcept
         $ unificationProofStepVariablesToCommon
             existingVars variableMapping1 rawSubstitutionProof
+
+    let
+        variablesInLeftAxiom = pureAllVariables axiomLeftRaw
+        toVariable (AxiomVariable v) = v
+        toVariable (ConfigurationVariable v) = v
+        substitutions =
+            Set.fromList . fmap (toVariable . fst) $ normalizedSubstitution
+    if Predicate.isFalse condition
```

Comment:
The first part of this condition, `Predicate.isFalse condition`, does not seem to be related to the error message below; is that right?